src/components/icon/icon-registry.ts

```diff
@@ -0,0 +1,370 @@
+   */
+  classNameForFontAlias(alias: string): string {
+    return this._fontCssClassesByAlias.get(alias) || alias;
+  }
+
+  /**
+   * Sets the CSS class name to be used for icon fonts when an <md-icon> component does not
+   * have a fontSet input value, and is not loading an icon by name or URL.
+   */
+  setDefaultFontSetClass(className: string) {
+    this._defaultFontSetClass = className;
+    return this;
+  }
+
+  /**
+   * Returns the CSS class name to be used for icon fonts when an <md-icon> component does not
+   * have a fontSet input value, and is not loading an icon by name or URL.
+   */
+  getDefaultFontSetClass(): string {
+    return this._defaultFontSetClass;
+  }
+
+  /**
+   * Returns an Observable that produces the icon (as an <svg> DOM element) from the given URL.
+   * The response from the URL may be cached so this will not always cause an HTTP request, but
+   * the produced element will always be a new copy of the originally fetched icon. (That is,
+   * it will not contain any modifications made to elements previously returned).
+   */
+  getSvgIconFromUrl(url: string): Observable<SVGElement> {
+    if (this._cachedIconsByUrl.has(url)) {
+      return Observable.of(this._cachedIconsByUrl.get(url).cloneNode(true));
+    }
+    return this._loadSvgIconFromConfig(new SvgIconConfig(url))
+        .do(svg => this._cachedIconsByUrl.set(url, svg))
+        .map(svg => svg.cloneNode(true));
+  }
+
+  /**
+   * Returns an Observable that produces the icon (as an <svg> DOM element) with the given name
+   * and namespace. The icon must have been previously registered with addIcon or addIconSet;
+   * if not, the Observable will throw an MdIconNameNotFoundException.
+   */
+  getNamedSvgIcon(name: string, namespace = ''): Observable<SVGElement> {
+    // Return (copy of) cached icon if possible.
+    const iconKey = namespace + ':' + name;
+    if (this._svgIconConfigs.has(iconKey)) {
+      return this._getSvgFromConfig(this._svgIconConfigs.get(iconKey));
+    }
+    // See if we have any icon sets registered for the namespace.
+    const iconSetConfigs = this._iconSetConfigs.get(namespace);
+    if (iconSetConfigs) {
+      return this._getSvgFromIconSetConfigs(name, this._iconSetConfigs.get(namespace));
+    }
+    return Observable.throw(new MdIconNameNotFoundException(iconKey));
+  }
+
+  /**
+   * Returns the cached icon for a SvgIconConfig if available, or fetches it from its URL if not.
+   */
+  private _getSvgFromConfig(config: SvgIconConfig): Observable<SVGElement> {
+    if (config.svgElement) {
+      // We already have the SVG element for this icon, return a copy.
+      return Observable.of(config.svgElement.cloneNode(true));
+    } else {
+      // Fetch the icon from the config's URL, cache it, and return a copy.
+      return this._loadSvgIconFromConfig(config)
+          .do((svg: SVGElement) => config.svgElement = svg)
```

Comment:
Is the explicit `SVGElement` type necessary here? I would have thought TS inferred it from the generic.